test/elements.test.ts

```diff
@@ -0,0 +1,44 @@
+import { ElementDictionary } from "../src/data/elements";
+import { Phases, Phase } from "../src/interfaces/element-props";
+
+function checkUrl(url: string): boolean {
+  let isUrl = true;
+  try {
+    new URL(url);
+  } catch (e) {
+    isUrl = false;
+  }
+  return isUrl;
+}
+
+function checkPhase(phase: Phase): boolean {
+  return Object.values(Phases).includes(phase);
+}
+
+describe("Periodic Table Elements", () => {
+  const elements: string[] = Object.keys(ElementDictionary);
+  it("should return true for correct URL of source", () => {
+    for (let el of elements) {
+      expect(checkUrl(ElementDictionary[el].source)).toBe(true);
+    }
+  });
+  it("should return true for correct URL of Spectral Image", () => {
+    for (let el of elements) {
+      if (typeof ElementDictionary[el].spectralImage === "string") {
+        expect(checkUrl(ElementDictionary[el].spectralImage as string)).toBe(true);
+      }
+    }
+  });
+  it("should return true for correct URL of Spectral Image", () => {
+    for (let el of elements) {
+      if (typeof ElementDictionary[el].spectralImage !== "string") {
+        expect(checkUrl(ElementDictionary[el].spectralImage as string)).toBe(false);
```

Comment:
This type assertion is not needed

Reply:
Correct @qur786  you are already doing the type check in if block so need need for assertion.

Reply:
@saikat-samanta @sumitnegi7  then i would need to declare the parameter of checkUrl function to be optional or combination of string | undefined.

Reply:
Nope if you are using `if (typeof ElementDictionary[el].spectralImage !== "string") {` then inside of the if block you have already narrowed down to the type string.  undefined is not possible for spectralImage inside this if block. If you are stilll not able to understand we can connect.

Reply:
@sumitnegi7  I thinks test case is not needed for those values.

Reply:
Lets connect

Reply:
@sumitnegi7  that's what I am saying , we don't need to define test case for those values. If we keep this test case for these values then we need to change checkUrl( ) parameter type , I have already filtered string values for sprectralImage link in the previous case .

Reply:
`I have already filtered string values for sprectralImage link in the previous case .`
That is correct i am not talking about that test case here you are passing the undefined value as string that is what i am saying. If you doing a check saying `(typeof ElementDictionary[el].spectralImage !== "string")` then inside of it the value is undefined as per the interface. So typecasting undefined as string is not a good practice. If you are unclear about my comment we can discuss this over call too.

Reply:
I clearly understand it @sumitnegi7  , that's why I asked you that if I should change the type of the parameter  type to string | undefined ,so that you we don't use "as string " assertion . or else why we will test those values for checkUrl() if they don't have any values at all.....

Reply:
It is not about as string assertion @qur786 you are basically sending undefined values and accepting them as string in checkUrl() function that is the issue.And the undefined value you are sending you are forcefully casting them as string.

Reply:
```js
import { ElementDictionary } from "../src/data/elements";
import { Phases } from "../src/interfaces/element-props";

function checkUrl(url?: string): boolean {
  let isUrl = true;
  try {
    typeof url === "string" ? new URL(url) : (isUrl = false);
  } catch (e) {
    isUrl = false;
  }
  return isUrl;
}

describe("Periodic Table Elements", () => {
  const elements = Object.keys(ElementDictionary);
  it("should return true for correct URL of source", () => {
    for (let el of elements) {
      const { source } = ElementDictionary[el];
      expect(checkUrl(source)).toBe(true);
    }
  });
  it("should return true for correct URL of Spectral Image", () => {
    for (let el of elements) {
      const { spectralImage } = ElementDictionary[el];
      if (typeof spectralImage === "string") {
        expect(checkUrl(spectralImage)).toBe(true);
      }
    }
  });
  it("should return true for correct URL of Spectral Image", () => {
    for (let el of elements) {
      const { spectralImage } = ElementDictionary[el];
      if (typeof spectralImage !== "string") {
        expect(checkUrl(spectralImage)).toBe(false);
      }
    }
  });
  it("should return false if the phase is among Gas ,Liquid or Solid", () => {
    for (let el of elements) {
      const { phase } = ElementDictionary[el];
      expect(Object.values(Phases).includes(phase)).toBe(true);
    }
  });
});

```

Reply:
That what I asked about yesterday sumit to make the parameter for checkUrl ( l)parameter to be string and optional .

Reply:
Okay no problem i should have suggested with code snippet earlier to give more clarity. Also look into destructuring too to me it looks more readable what do you think?

Reply:
```ts
function checkUrl(url?: string): boolean {
  let isUrl = true;
  try {
    typeof url === "string" ? new URL(url) : (isUrl = false);
  } catch (e) {
    isUrl = false;
  }
  return isUrl;
}
```

Is this change made?

Reply:
No sir , I was waiting for your opinion on it

Reply:
```ts
function checkUrl(url?: string): boolean {
  let isUrl = false;
  if (typeof url === "string") {
    try {
      new URL(url);
      isUrl = true;
    } catch (e) {
    }
  }
  return isUrl;
}
```

Lets make the change